core/src/main/java/com/linecorp/armeria/server/http/dynamic/HttpDynamicService.java

```diff
@@ -0,0 +1,118 @@
+
+    /**
+     * Creates a new {@link HttpDynamicService} with the specified strategy {@code object}.
+     * */
+    public static HttpDynamicService forStrategy(Object object) {
+        List<Entry> entries = Lists.newArrayList();
+
+        for (Method method : object.getClass().getMethods()) {
+            try {
+                DynamicMethod dynamicMethod = DynamicMethod.of(object, method);
+                RequestMapping requestMapping = method.getAnnotation(RequestMapping.class);
+                if (null == requestMapping) {
+                    throw new IllegalArgumentException("Not annotated with RequestMapping annotation.");
+                }
+                DynamicPath dynamicPath = DynamicPath.of(requestMapping.value(),
+                                                         dynamicMethod.getParameterNames());
+                entries.add(new Entry(dynamicPath, dynamicMethod));
+            } catch (IllegalArgumentException e) {
+                logger.warn("Ill-configured method {}: {}", method.getName(), e.getMessage());
+            }
+        }
+
+        return new HttpDynamicService(entries);
+    }
+
+    /**
+     * Dynamic invocation entry, with its own dynamic (or bindable) path and target method.
+     */
+    private static class Entry {
+        private DynamicPath path;
+        private DynamicMethod method;
+
+        Entry(DynamicPath path, DynamicMethod method) {
+            this.path = path;
+            this.method = method;
+        }
+
+        public boolean matches(String mappedPath) {
+            return path.matches(mappedPath);
+        }
+
+        public Object invoke(String mappedPath) throws InvocationTargetException, IllegalAccessException {
+            Map<String, String> bound = path.bind(mappedPath);
+            return method.invoke(bound);
+        }
+    }
+
+    private List<Entry> entries;
+
+    private HttpDynamicService(List<Entry> entries) {
+        this.entries = entries;
+    }
+
+    @Override
+    public HttpResponse serve(ServiceRequestContext ctx, HttpRequest req) throws Exception {
+        final DefaultHttpResponse res = new DefaultHttpResponse();
+        final long current = System.currentTimeMillis();
+        String path = ctx.mappedPath();
+
+        Optional<Entry> invokable = entries.stream().filter(e -> e.matches(path)).findFirst();
+
+        if (!invokable.isPresent()) {
+            res.respond(HttpStatus.NOT_FOUND);
+        } else {
+            try {
+                Object returned = invokable.get().invoke(path);
+                final HttpData data = HttpData.ofUtf8(returned.toString());
+
+                HttpHeaders headers = HttpHeaders.of(HttpStatus.OK)
+                                                 .set(HttpHeaderNames.CONTENT_TYPE,
+                                                      MediaType.PLAIN_TEXT_UTF_8.type())
+                                                 .set(HttpHeaderNames.CONTENT_ENCODING, Charsets.UTF_8.name())
```

Comment:
`content-encoding` header is not where you set the character set of the content. Let's remove it.
https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Encoding